### PR TITLE
Fix `camp.path` by making generated RegExps only match entire inputs

### PR DESCRIPTION
**Pull request by jankeromnes**
The problem:

    server.path('foo/:bar', function (req, res) {
      res.end('bar=' + req.data.bar);
    });
    server.path('foo/:bar/baz', function (req, res) {
      res.end('baz');
    });
    http.get('http://localhost/foo/quux/baz', function (res) {
      res.on('data', function (data) {
        // FIXME: `data` is 'bar=quux' instead of 'baz'.
        // There is no way to get 'baz', the route is unreachable.
      });
    });

The cause: `regexFromString` returns a partially matching regex (without `^` or `$`), so the path `'foo/:bar'` catches anything from `'not/foo/:bar'` to `'foo/:bar/baz'`.

The solution: Make `regexFromString` return a fully matching regex.

I implemented the solution, and contributed a test that fails when the fix is not applied.

### CODE REVIEW

File: lib/camp.js [hunk 1]
```diff
@@ -351,6 +351,7 @@ function SecureCamp(opts) {
 inherits(SecureCamp, https.Server);
 
 Camp.prototype.handle = SecureCamp.prototype.handle =
+Camp.prototype.use = SecureCamp.prototype.use =
```
Comment (espadrine) [inline lib/camp.js]: Is it really useful to be an express lookalike? Especially considering the subtle differences, I find that unnecessary, borderline error-inducing.

Regardless of the decision to add it, it would need to be a separate patch.

**Comment by jankeromnes** [comment]
Thanks! Please feel free to rebase my commits (maybe except the use/handle alias) on top of master, or I can do it.

**Comment by jankeromnes** [comment]
Pushed the two accepted commits 9c184069314aba1380dd37a546db9db19aaeb78e and 754b05702e57b997a909f53231ba327af7c7f8f0.
